### ISSUE TITLE
API usage statistics fluctuating

### ISSUE THREAD

**Issue by GlennBoschmans** [wso2/analytics-apim#316]
While doing a PoC for a customer I've noticed that the API usage statistics are fluctuating. When I check the usage data in the API_REQUEST_SUMMARY table the total request count for an API changes, having an impact on the statistics shown in the Store and the Publisher. 

This issue happens on different types of set-up. I've noticed this when API Manager 2.0.0 and the Analytics were installed on the same machine using H2 database. It also happened when I switched to a MySql database. 
I now have an installation where APIM 2.0.0 and the Analytics run on a different server, using MySql as database.

In the example below you can see that today (August 31) the total amount of requests on August 29 changes. 

When I executed the query the first time the total amount of requests for the Monuments 1.0.0 API by user monuments_client is 10 (6 + 4).

```
mysql> select * from API_REQUEST_SUMMARY;
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| api       | api_version               | version | apiPublisher | consumerKey                  | userId                        | context          | max_request_time | total_request_count | hostName      | year | month | day | time             |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472479494000 |                   6 | 192.168.36.72 | 2016 |     8 |  29 | 2016-08-29 16:04 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472626117000 |                   1 | 192.168.36.72 | 2016 |     8 |  31 | 2016-08-31 08:48 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472477680000 |                   4 | wso2-openapi2 | 2016 |     8 |  29 | 2016-08-29 15:34 |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
3 rows in set (0.00 sec)
```

About ten minutes later (after doing some new requests, on August 31) the total of requests is 7 (6 + 1). 

```
mysql> select * from API_REQUEST_SUMMARY;
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| api       | api_version               | version | apiPublisher | consumerKey                  | userId                        | context          | max_request_time | total_request_count | hostName      | year | month | day | time             |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472479494000 |                   6 | 192.168.36.72 | 2016 |     8 |  29 | 2016-08-29 16:04 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472626272000 |                   2 | 192.168.36.72 | 2016 |     8 |  31 | 2016-08-31 08:51 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472475259000 |                   1 | wso2-openapi2 | 2016 |     8 |  29 | 2016-08-29 14:54 |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
3 rows in set (0.00 sec)
```

Again some 20 minutes later, the amount is 8 (4+4).

```
mysql> select * from API_REQUEST_SUMMARY;
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| api       | api_version               | version | apiPublisher | consumerKey                  | userId                        | context          | max_request_time | total_request_count | hostName      | year | month | day | time             |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472479494000 |                   4 | 192.168.36.72 | 2016 |     8 |  29 | 2016-08-29 16:04 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472626117000 |                   1 | 192.168.36.72 | 2016 |     8 |  31 | 2016-08-31 08:48 |
| Monuments | creator--Monuments:v1.0.0 | 1.0.0   | creator      | IYkOkwfISxpVHvH596DWfxLTErca | monuments_client@carbon.super | /monuments/1.0.0 |    1472477680000 |                   4 | wso2-openapi2 | 2016 |     8 |  29 | 2016-08-29 15:34 |
+-----------+---------------------------+---------+--------------+------------------------------+-------------------------------+------------------+------------------+---------------------+---------------+------+-------+-----+------------------+
3 rows in set (0.00 sec)
```

**Comment by angusfz** [comment]
+1, got the same wired situation.
Im usign wso2am-2.0.0 + wso2am-analytics-2.0.0.
And, i found that today's request_count will add to other day.
Anyone can help ?

**Comment by fazlan-nazeem** [comment]
we have identified and fixed several issues as such and these issues should be resolved from 2.2.0 release onwards. Hence closing the issue.
